[Logs / Metrics UI] Client side NP migration

Kerry350 opened the pull request:
## Summary

This is phase one of the client side new platform migration of Logs and Metrics.

## What's been done?

- Sets up the main shim layer (e.g. an `index` and `plugin` file that work in the same way as the real new platform, taking in and using `core` and `plugins`).

- Migrates several of our `ui/***` imports to use new platform services, this is via consumption of `core` and `plugins` via the `<KibanaContextProvider />`:
  - `ui/capabilities` ✅
  - `ui/chrome` ☑️
  - `ui/kfetch` ✅ (with caveat *)
  - `ui/notify` ✅
  - `ui/index_patterns` ✅
  - `ui/i18n` ✅

✅ = fully migrated ☑️ = some references left

### * `kfetch` caveat

On the metrics side `ui/kfetch` has been removed 100%, and `core.http.fetch` is being used for all requests via the `useHTTPRequest` hook. On the logs side this is a little trickier to migrate (not "hard" but a lot of changes), and I would like to do this in a followup ticket / PR.

The problem here is that the low level `useTrackedPromise` hook is being used, and not the  high level `useHTTPRequest` hook. Within many of these uses of `useTrackedPromise` the `createPromise` method calls `callXYZ` functions which are imported from separate files. These functions, housed in separate files, import `ui/kfetch` directly at that level. As these aren't hooks or components themselves, but just functions, we can't access `http.fetch` from the Kibana context. It would need to be passed in as a parameter.

I would like to migrate the logs uses over to the `useHTTPRequest` hook. This keeps patterns the same between the two apps, and (most importantly) keeps acess to `core.http.fetch` in one location only.

For now Logs is using `http.fetch` imported from `ui/new_platform` but this is only a temporary solution.

## What's left?

- The remaining `ui/***` imports:
  - `ui/vis/lib/timezone` (1 ref)
  - `ui/utils/query_string` (3 refs)
  - `ui/new_platform` (this is mostly because of the `kfetch` caveat above, but we want to remove these remaining `ui/new_platform` references before the final switchover) (15 refs)
  - `ui/registry/feature_catalogue` (1 ref)

- The real switchover to the NP (moving directories, removing `app.ts`, having `index` export `plugin` etc).

## Improvements

Right now access to servives via the `KibanaContextProvider` is very direct, e.g. `useKibana().services.application?.capabilities;`. We may still want to wrap each service that we use in our own hook (e.g. `useCapabilities` in this example, which would call `useKibana` internally), so that if Kibana services / APIs change later down the road we can easily change them in one place, instead of many. Personally, this is something I would like to see (in a followup).

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15071/)
* Commit: d21691ae85d8bbfabbb631c04a56fbe30da327b6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15071/","number":"15071","commit":"d21691ae85d8bbfabbb631c04a56fbe30da327b6"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15085/)
* Commit: 3b8f4af1d6c7782d079290d44271e1a7b7b128f0

### History
* :broken_heart: [Build #15071](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15071/) failed d21691ae85d8bbfabbb631c04a56fbe30da327b6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15071/","number":"15071","commit":"d21691ae85d8bbfabbb631c04a56fbe30da327b6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15085/","number":"15085","commit":"3b8f4af1d6c7782d079290d44271e1a7b7b128f0"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15524/)
* Commit: 74e76819383f11e85844f19a51ea78b8421d204f

### History
* :broken_heart: [Build #15085](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15085/) failed 3b8f4af1d6c7782d079290d44271e1a7b7b128f0
* :broken_heart: [Build #15071](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15071/) failed d21691ae85d8bbfabbb631c04a56fbe30da327b6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15071/","number":"15071","commit":"d21691ae85d8bbfabbb631c04a56fbe30da327b6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15085/","number":"15085","commit":"3b8f4af1d6c7782d079290d44271e1a7b7b128f0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15524/","number":"15524","commit":"74e76819383f11e85844f19a51ea78b8421d204f"}]}
PIPELINE-->

jasonrhodes commented on x-pack/legacy/plugins/infra/public/components/metrics_explorer/chart.tsx:
```diff
@@ -49,132 +48,129 @@ interface Props {
+  timeRange,
+  onTimeChange,
+}: Props) => {
+  const uiCapabilities = useKibana().services.application?.capabilities;
```
`?.` nice!!

jasonrhodes commented on x-pack/legacy/plugins/infra/public/components/metrics_explorer/chart.tsx:
```diff
@@ -18,8 +18,6 @@ import {
 } from '@elastic/charts';
```
Most of the changes in this file look like indentation, for the most part, right?

Kerry350 replied:
Yeah, it's the same for any file where `injectUICapabilities` was removed.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15712/)
* Commit: 593274df45db7b4dce6c0d61b7508b4a03bc4f56

### History
* :broken_heart: [Build #15524](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15524/) failed 74e76819383f11e85844f19a51ea78b8421d204f
* :broken_heart: [Build #15085](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15085/) failed 3b8f4af1d6c7782d079290d44271e1a7b7b128f0
* :broken_heart: [Build #15071](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15071/) failed d21691ae85d8bbfabbb631c04a56fbe30da327b6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15071/","number":"15071","commit":"d21691ae85d8bbfabbb631c04a56fbe30da327b6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15085/","number":"15085","commit":"3b8f4af1d6c7782d079290d44271e1a7b7b128f0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15524/","number":"15524","commit":"74e76819383f11e85844f19a51ea78b8421d204f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15712/","number":"15712","commit":"593274df45db7b4dce6c0d61b7508b4a03bc4f56"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15734/)
* Commit: 70609c1724837ac5d3493050155f44ec62a6ea2e

### History
* :broken_heart: [Build #15712](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15712/) failed 593274df45db7b4dce6c0d61b7508b4a03bc4f56
* :broken_heart: [Build #15524](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15524/) failed 74e76819383f11e85844f19a51ea78b8421d204f
* :broken_heart: [Build #15085](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15085/) failed 3b8f4af1d6c7782d079290d44271e1a7b7b128f0
* :broken_heart: [Build #15071](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15071/) failed d21691ae85d8bbfabbb631c04a56fbe30da327b6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15071/","number":"15071","commit":"d21691ae85d8bbfabbb631c04a56fbe30da327b6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15085/","number":"15085","commit":"3b8f4af1d6c7782d079290d44271e1a7b7b128f0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15524/","number":"15524","commit":"74e76819383f11e85844f19a51ea78b8421d204f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15712/","number":"15712","commit":"593274df45db7b4dce6c0d61b7508b4a03bc4f56"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15734/","number":"15734","commit":"70609c1724837ac5d3493050155f44ec62a6ea2e"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15841/)
* Commit: 85e62ec870235fda9eba4f9c4828fc58a9fba1f1

### History
* :broken_heart: [Build #15734](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15734/) failed 70609c1724837ac5d3493050155f44ec62a6ea2e
* :broken_heart: [Build #15712](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15712/) failed 593274df45db7b4dce6c0d61b7508b4a03bc4f56
* :broken_heart: [Build #15524](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15524/) failed 74e76819383f11e85844f19a51ea78b8421d204f
* :broken_heart: [Build #15085](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15085/) failed 3b8f4af1d6c7782d079290d44271e1a7b7b128f0
* :broken_heart: [Build #15071](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15071/) failed d21691ae85d8bbfabbb631c04a56fbe30da327b6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15071/","number":"15071","commit":"d21691ae85d8bbfabbb631c04a56fbe30da327b6"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15085/","number":"15085","commit":"3b8f4af1d6c7782d079290d44271e1a7b7b128f0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15524/","number":"15524","commit":"74e76819383f11e85844f19a51ea78b8421d204f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15712/","number":"15712","commit":"593274df45db7b4dce6c0d61b7508b4a03bc4f56"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15734/","number":"15734","commit":"70609c1724837ac5d3493050155f44ec62a6ea2e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15841/","number":"15841","commit":"85e62ec870235fda9eba4f9c4828fc58a9fba1f1"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16037/)
* Commit: 9736891675f8af5a7b8141464833bbae9f85ca21

### History
* :broken_heart: [Build #15841](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15841/) failed 85e62ec870235fda9eba4f9c4828fc58a9fba1f1
* :broken_heart: [Build #15734](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15734/) failed 70609c1724837ac5d3493050155f44ec62a6ea2e
* :broken_heart: [Build #15712](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15712/) failed 593274df45db7b4dce6c0d61b7508b4a03bc4f56
* :broken_heart: [Build #15524](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15524/) failed 74e76819383f11e85844f19a51ea78b8421d204f
* :broken_heart: [Build #15085](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15085/) failed 3b8f4af1d6c7782d079290d44271e1a7b7b128f0

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15085/","number":"15085","commit":"3b8f4af1d6c7782d079290d44271e1a7b7b128f0"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15524/","number":"15524","commit":"74e76819383f11e85844f19a51ea78b8421d204f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15712/","number":"15712","commit":"593274df45db7b4dce6c0d61b7508b4a03bc4f56"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15734/","number":"15734","commit":"70609c1724837ac5d3493050155f44ec62a6ea2e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15841/","number":"15841","commit":"85e62ec870235fda9eba4f9c4828fc58a9fba1f1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16037/","number":"16037","commit":"9736891675f8af5a7b8141464833bbae9f85ca21"}]}
PIPELINE-->

spalger commented:
@elasticmachine merge upstream

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16264/)
* Commit: 843e7095f45504f8d3c2591bc8d35f41b303237b

### History
* :broken_heart: [Build #16037](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16037/) failed 9736891675f8af5a7b8141464833bbae9f85ca21
* :broken_heart: [Build #15841](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15841/) failed 85e62ec870235fda9eba4f9c4828fc58a9fba1f1
* :broken_heart: [Build #15734](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15734/) failed 70609c1724837ac5d3493050155f44ec62a6ea2e
* :broken_heart: [Build #15712](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15712/) failed 593274df45db7b4dce6c0d61b7508b4a03bc4f56
* :broken_heart: [Build #15524](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15524/) failed 74e76819383f11e85844f19a51ea78b8421d204f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15524/","number":"15524","commit":"74e76819383f11e85844f19a51ea78b8421d204f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15712/","number":"15712","commit":"593274df45db7b4dce6c0d61b7508b4a03bc4f56"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15734/","number":"15734","commit":"70609c1724837ac5d3493050155f44ec62a6ea2e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15841/","number":"15841","commit":"85e62ec870235fda9eba4f9c4828fc58a9fba1f1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16037/","number":"16037","commit":"9736891675f8af5a7b8141464833bbae9f85ca21"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16264/","number":"16264","commit":"843e7095f45504f8d3c2591bc8d35f41b303237b"}]}
PIPELINE-->

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16390/)
* Commit: c31e6cf0625d982f917cc2aab61e91904be8ccac

### History
* :broken_heart: [Build #16264](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16264/) failed 843e7095f45504f8d3c2591bc8d35f41b303237b
* :broken_heart: [Build #16037](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16037/) failed 9736891675f8af5a7b8141464833bbae9f85ca21
* :broken_heart: [Build #15841](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15841/) failed 85e62ec870235fda9eba4f9c4828fc58a9fba1f1
* :broken_heart: [Build #15734](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15734/) failed 70609c1724837ac5d3493050155f44ec62a6ea2e
* :broken_heart: [Build #15712](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15712/) failed 593274df45db7b4dce6c0d61b7508b4a03bc4f56

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15712/","number":"15712","commit":"593274df45db7b4dce6c0d61b7508b4a03bc4f56"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15734/","number":"15734","commit":"70609c1724837ac5d3493050155f44ec62a6ea2e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15841/","number":"15841","commit":"85e62ec870235fda9eba4f9c4828fc58a9fba1f1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16037/","number":"16037","commit":"9736891675f8af5a7b8141464833bbae9f85ca21"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16264/","number":"16264","commit":"843e7095f45504f8d3c2591bc8d35f41b303237b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16390/","number":"16390","commit":"c31e6cf0625d982f917cc2aab61e91904be8ccac"}]}
PIPELINE-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16449/)
* Commit: d9c874d387010637632b0ccfaf69c2f73d9e033b

### History
* :broken_heart: [Build #16390](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16390/) failed c31e6cf0625d982f917cc2aab61e91904be8ccac
* :broken_heart: [Build #16264](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16264/) failed 843e7095f45504f8d3c2591bc8d35f41b303237b
* :broken_heart: [Build #16037](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16037/) failed 9736891675f8af5a7b8141464833bbae9f85ca21
* :broken_heart: [Build #15841](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15841/) failed 85e62ec870235fda9eba4f9c4828fc58a9fba1f1
* :broken_heart: [Build #15734](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15734/) failed 70609c1724837ac5d3493050155f44ec62a6ea2e

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15734/","number":"15734","commit":"70609c1724837ac5d3493050155f44ec62a6ea2e"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15841/","number":"15841","commit":"85e62ec870235fda9eba4f9c4828fc58a9fba1f1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16037/","number":"16037","commit":"9736891675f8af5a7b8141464833bbae9f85ca21"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16264/","number":"16264","commit":"843e7095f45504f8d3c2591bc8d35f41b303237b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16390/","number":"16390","commit":"c31e6cf0625d982f917cc2aab61e91904be8ccac"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16449/","number":"16449","commit":"d9c874d387010637632b0ccfaf69c2f73d9e033b"}]}
PIPELINE-->

jasonrhodes commented on x-pack/legacy/plugins/infra/public/components/waffle/node_context_menu.tsx:
```diff
@@ -29,102 +28,99 @@ interface Props {
+  // We need to have some exceptions until 7.0 & ECS is finalized. Reference
+  // #26620 for the details for these fields.
+  // TODO: This is tech debt, remove it after 7.0 & ECS migration.
+  const apmField = nodeType === InfraNodeType.host ? 'host.hostname' : inventoryModel.fields.id;
```
@simianhacker do you have context about this comment? Can we remove something and get rid of this comment now that we're post 7.0?

jasonrhodes commented on x-pack/test/functional/apps/infra/link_to.ts:
```diff
@@ -22,7 +22,7 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
       };
       const expectedSearchString =
-        "logFilter=(expression:'trace.id:433b4651687e18be2c6c8e3b11f53d09',kind:kuery)&logPosition=(position:(tiebreaker:0,time:1565707203194))&sourceId=default&_g=()";
+        "logFilter=(expression:'trace.id:433b4651687e18be2c6c8e3b11f53d09',kind:kuery)&logPosition=(position:(tiebreaker:0,time:1565707203194))&sourceId=default";
```
Is this change expected?

Kerry350 replied:
It is insofar as me making it, but I don't understand the change in behaviour. This functional test is testing `pages/link_to/redirect_to_logs`, in that component we only add the expected query parameters (`logPosition` etc). 

I know `_g=()` is something added globally by Kibaba, but I don't really understand _when_. We don't read from or write to `_g=()` in our apps.  Within this component we are now within our app, and our react-router context. I guess the old behaviour was `&_g=()` was set at some point in the global angular router, and then our stuff just merged with it. And now it's just the stuff we actually set ourselves.

Ultimately, I don't know where the difference in behaviour comes from. As I haven't changed the uses of `ui/utils/query_string` or anything to do with URLs.

However, this new behaviour is more inline with what I'd expect (the URL only contains what we set and use). Would be nice to understand the root change, though.

Kerry350 replied:
Oh - actually, as a guess, maybe it's because we removed all that code that accessed Angular and did those buffer calls, and we're just mounting straight within the "React world" now.

jasonrhodes replied:
Ohhh that's a good guess, and I like the new behavior too. Sounds safe.

Kerry350 commented:
@jasonrhodes Thanks for the review 🙏 

I did test the ones within the `useHTTPRequest` hook. But not the other use. I'll give that a test now, if there aren't any problems I'll merge.

Kerry350 commented:
@jasonrhodes Good shout on the notifications stuff. There was nothing broken as such but the timeouts were way too quick, so I've added a 3 second delay on those. Also, with the handling in `useHTTPRequest` the `Error` instance doesn't seem to be a `HttpFetchError` with `response` (etc) properties as I expected - so I've added a fallback to display just the `message` if necessary. I will look into this today, but couldn't find anything after 10 minutes of initial poking. I didn't want the formatting of a rarely seen message to hold up merging.

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16683/)
* Commit: 56fbd187a144ebcd7d187dd2db94f0da7da9b0aa

### History
* :green_heart: [Build #16449](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16449/) succeeded d9c874d387010637632b0ccfaf69c2f73d9e033b
* :broken_heart: [Build #16390](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16390/) failed c31e6cf0625d982f917cc2aab61e91904be8ccac
* :broken_heart: [Build #16264](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16264/) failed 843e7095f45504f8d3c2591bc8d35f41b303237b
* :broken_heart: [Build #16037](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16037/) failed 9736891675f8af5a7b8141464833bbae9f85ca21
* :broken_heart: [Build #15841](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15841/) failed 85e62ec870235fda9eba4f9c4828fc58a9fba1f1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/15841/","number":"15841","commit":"85e62ec870235fda9eba4f9c4828fc58a9fba1f1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16037/","number":"16037","commit":"9736891675f8af5a7b8141464833bbae9f85ca21"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16264/","number":"16264","commit":"843e7095f45504f8d3c2591bc8d35f41b303237b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16390/","number":"16390","commit":"c31e6cf0625d982f917cc2aab61e91904be8ccac"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16449/","number":"16449","commit":"d9c874d387010637632b0ccfaf69c2f73d9e033b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/16683/","number":"16683","commit":"56fbd187a144ebcd7d187dd2db94f0da7da9b0aa"}]}
PIPELINE-->
